File: problem/Longest_Subarray_Delete/longestSubarray.c
int longestSubarray(int *nums, int numsSize)
{
    int left = 0, zeros = 0, res = 0;

    for (int right = 0; right < numsSize; right++)
    {
        if (nums[right] == 0)
            zeros++;

        while (zeros > 1)
        {
            if (nums[left] == 0)
                zeros--;
            left++;
        }

        int length = right - left;
        if (length > res)
            res = length;
    }

    return res;
}

File: problem/Longest_Subarray_Delete/longestSubarray.py
class Solution:
    def longestSubarray(self, nums):
        left = 0
        zeros = 0
        res = 0
        
        for right in range(len(nums)):
            if nums[right] == 0:
                zeros += 1
            
            while zeros > 1:
                if nums[left] == 0:
                    zeros -= 1
                left += 1
            
            res = max(res, right - left)
        
        return res
